tls/s2n_cipher_suites.c

```diff
@@ -1227,15 +1228,27 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
                 continue;
             }
 
+            /* For TLS1.3 when PSKs are present, the server must consider the hash algorithm associated with the chosen PSK,
+             * when choosing a cipher suite. Server MUST reject any cipher suite without a matching hash algorithm and 
+             * continue to the next candidate. 
+             * */     
+            if (conn->actual_protocol_version >= S2N_TLS13 && conn->psk_params.chosen_psk != NULL) {
+                s2n_hmac_algorithm chosen_psk_hmac_alg;
+                GUARD(s2n_hash_hmac_alg(conn->psk_params.chosen_psk->hash_alg, &chosen_psk_hmac_alg));
```

Comment:
Didn't we discuss switching the connection field to s2n_hmac_algorithm? We convert the s2n_hash_algorithm to an s2n_hmac_algorithm every time we use it

Reply:
Decided to do it in a separate PR and not include it in the cipher selection logic changes. Created this issue to track it https://github.com/awslabs/s2n/issues/2458.